driver-core/src/main/com/mongodb/client/model/search/SearchOperator.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.search;
+
+import com.mongodb.annotations.Immutable;
+import com.mongodb.lang.Nullable;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.conversions.Bson;
+
+import java.util.List;
+
+/**
+ *
+ * @since 4.4
+ */
+@Immutable
+public abstract class SearchOperator implements Bson {
```

Comment:
Base class for all search operators.  It doesn't pull a lot of weight...  but I want it to be an abstract class rather than an interface so that we can lock down the ability to subclass by keeping the constructor non-public.